### PR TITLE
feat: added integration with d3fc-label-layout

### PR DESCRIPTION
**Pull request by ColinEberhardt**
BREAKING CHANGE: this removes the 'rectangles' layout component replacing it with the equivalent 'label' layout component from d3fc-label-layout

This integrated the  d3fc-label-layout module using our existing d3fc.io website structure. As we break things up we are going to have to review the 'component' navigation again. We could do something similar to assemble (http://assemble.io) (sorry for mentioning that SSG again!), where it shows a brief summary 'card' of the various sub-projects.

### CODE REVIEW

File: site/src/components/layout/rectangles-example-measure.js [hunk 1]
```diff
@@ -114,15 +114,10 @@ var data = [
 //START
 var labelPadding = 2;
 
-// the container component is used to add padding around a text label
-var label = fc.tool.container()
+// a component which renders the labels
+var label = fc.layout.textLabel()
```
Comment (chrisprice) [inline site/src/components/layout/rectangles-example-measure.js]: The naming seems forced to fit in with `fc.layout.label`, not that I can think of a better name...

File: src/layout/label.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { range } from '../util/scale';
+import { rebindAll } from 'd3fc-rebind';
+import { label as labelLayout } from 'd3fc-label-layout';
+
+export default function(strategy) {
+
+    var adaptee = labelLayout(strategy);
+
+    var xScale = null,
```
Comment (chrisprice) [inline src/layout/label.js]: Normally we'd default the scales to e.g. `d3.scale.identitiy`.

File: src/layout/label.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        yScale = null;
+
+
+    var label = function(selection) {
+        // automatically set the bounds of the strategy based on the scale range
+        if (strategy && strategy.bounds && xScale && yScale) {
```
Comment (chrisprice) [inline src/layout/label.js]: Does it ever make sense for this component to be used without setting scales?

File: src/layout/label.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            var xRange = range(xScale),
+                yRange = range(yScale);
+            strategy.bounds([
```
Comment (chrisprice) [inline src/layout/label.js]: Just noticed that bounds only accepts two arguments, I assume that means it's assuming zero as the origin? Maybe that should be changed (separately!)

File: site/src/components/layout/rectangles-example-measure.js (hunk 1 above)
Comment (ColinEberhardt) [inline site/src/components/layout/rectangles-example-measure.js]: Yeah, I know - I didn't like it much either. I find myself writing code like this:

```
var labelLayout = fc.layout.label()
var label = fc.layout.textLabel()
```

Feels a bit confused.

**Comment by chrisprice** [comment]
A few minor comments but LGTM. Nice to see so much code deleted (or moved)!

File: src/layout/label.js (hunk 3 above)
Comment (ColinEberhardt) [inline src/layout/label.js]: That's a good question, and why I defaulted to `null` above. I think the answer is 'no', it doesn't make sense to use it without scale - in that case you just use the non-adapted layout. I'll simplify this code a little ...

**Comment by ColinEberhardt** [comment]
Thanks for the feedback - I've addressed these comments.

**Pull request merged**